### DISCUSSION TITLE
Replace pip to python3 -m pip

### DISCUSSION THREAD

**Discussion by lorddaedra** [fastapi/fastapi#8074] [Questions] [answered]
Replace `pip` to `python3 -m pip`

**Comment by Mdslino** [comment]
Why?

**Comment by lorddaedra** [comment]
Not a bug really, just perfectionism.

Same with pyvenv:
`Note The pyvenv script has been deprecated as of Python 3.6 in favor of using python3 -m venv to help prevent any potential confusion as to which Python interpreter a virtual environment will be based on.
`
https://docs.python.org/3/library/venv.html

All examples from https://docs.python.org/3/installing/index.html#basic-usage follow same scheme.

**Comment by HacKanCuBa** [comment]
Change it where exactly?

**Comment by purificant** [comment]
While `python3 -m pip` is more precise and removes ambiguity about which python version is used, `pip` is much shorter to type and makes a better candidate for user facing docs for example. I'd still use `python3 -m pip` in scripts that can run in multiple environments.

We are moving to a world where py3 is the only supported version given that py2 is end of life 1st January 2020, because of this python version ambiguity is a lot less of an issue going forward.

**Comment by tiangolo** [comment, marked as answer]
Thanks everyone for the discussion!

The preference of `pip install`, `python -m pip install`, `python3 -m pip install`, `pip3 install`, etc, is quite subjective and varies a lot depending on personal/team taste.

So, I prefer to leave the simplest version (`pip install`), and let everyone chose from that what to use.

That's the same reason why I don't suggest Pipenv, Poetry, `venv`, etc.

...Also, agree with @purificant 😅

---

I'll close this now, but feel free to add more comments or create new issues.

**Comment by lorddaedra** [comment]
I prefer `poetry add ...` now :-)
